test/extended/openstack/cloud-provider-config.go

```diff
@@ -104,6 +104,70 @@ var _ = g.Describe("[sig-installer][Suite:openshift/openstack] The Openshift", f
 				}
 			}
 		})
+
+		//https://bugzilla.redhat.com/show_bug.cgi?id=2074471
+		g.It("should update use-octavia property depending on the NetworkType", func() {
+
+			userConfigMap := config_params{
+				namespace: "openshift-config",
+				name:      "cloud-provider-config",
+				key:       "config",
+				skip:      nil,
+			}
+
+			ccmConfigMap := config_params{
+				namespace: "openshift-cloud-controller-manager",
+				name:      "cloud-conf",
+				key:       "cloud.conf",
+				skip:      nil,
+			}
+
+			section_name := "LoadBalancer"
+			property_name := "use-octavia"
+
+			cfg, err := e2e.LoadConfig()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			c, err := configv1client.NewForConfig(cfg)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			network, err := c.ConfigV1().Networks().Get(context.Background(), "cluster", metav1.GetOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			user_cfg, err := getConfig(clientSet,
+				userConfigMap.namespace,
+				userConfigMap.name,
+				userConfigMap.key)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			ccm_cfg, err := getConfig(clientSet,
+				ccmConfigMap.namespace,
+				ccmConfigMap.name,
+				ccmConfigMap.key)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			user_property_value, err := getPropertyValue(section_name, property_name, user_cfg)
+			o.Expect(err).NotTo(o.HaveOccurred(), "Error getting the property %q in section %q on configMap "+
+				"%q (namespace: %q)", property_name, section_name, userConfigMap.name, userConfigMap.namespace)
+
+			ccm_property_value, err := getPropertyValue(section_name, property_name, ccm_cfg)
+			o.Expect(err).NotTo(o.HaveOccurred(), "Error getting the property %q in section %q on configMap "+
+				"%q (namespace: %q)", property_name, section_name, ccmConfigMap.name, ccmConfigMap.namespace)
+
+			e2e.Logf("Checking config for networkType detected: %q", network.Status.NetworkType)
+			switch network.Status.NetworkType {
+			case "Kuryr":
+				o.Expect(ccm_property_value).Should(o.Equal("true"),
+					"Unexpected value for property %q on section %q. configMap '%q', namespace '%q' with Kuryr NetworkType.",
+					property_name, section_name, ccmConfigMap.name, ccmConfigMap.namespace)
+				e2e.Logf("  - Property %q with correct value %q on section %q in configMap %q (namespace: %q) with Kuryr NetworkType.",
+					property_name, ccm_property_value, section_name, ccmConfigMap.name, ccmConfigMap.namespace)
+			default:
+				o.Expect(ccm_property_value).Should(o.Equal(user_property_value),
+					"Unexpected value for property %q on section %q. configMap '%q', namespace '%q'.",
+					property_name, section_name, ccmConfigMap.name, ccmConfigMap.namespace)
+				e2e.Logf("  - Property %q with correct value %q on section %q in configMap %q (namespace: %q).",
+					property_name, ccm_property_value, section_name, ccmConfigMap.name, ccmConfigMap.namespace)
+			}
```

Comment:
Why not make it simple?

```
			e2e.Logf("Checking config for networkType detected: %q", network.Status.NetworkType)
			expected := userPropertyValue
			if network.Status.NetworkType == "Kuryr" {
				expected = "true"
			}

			o.Expect(ccmPropertyValue).Should(o.Equal(expected),
				"Unexpected value for property %q on section %q. configMap '%q', namespace '%q' with NetworkType=%s.",
				propertyName, sectionName, ccmConfigMap.name, ccmConfigMap.namespace, network.Status.NetworkType)
			e2e.Logf("  - Property %q with correct value %q on section %q in configMap %q (namespace: %q) with NetworkType=%s.",
				propertyName, ccmPropertyValue, sectionName, ccmConfigMap.name, ccmConfigMap.namespace)
```

Reply:
Thanks for the suggestion. I changed the logic a little bit. I included also the "enabled" property in the checks.